geode-core/src/main/java/org/apache/geode/internal/InternalDataSerializer.java

```diff
@@ -2728,14 +2729,69 @@ private static String readStringUTFFromDataInput(DataInput in) throws IOExceptio
     return in.readUTF();
   }
 
+  /**
+   * If not null then this field can be used to
+   * directly set the char[] "value" field on a String instance.
+   */
+  static final Field stringValueField;
+  static {
+    Field tempField;
+    try {
+      tempField = String.class.getDeclaredField("value");
```

Comment:
Prefer the package protected `String(char[] value, boolean share)` constructor. Constructing `String()` results in the allocation of a `char[0]`. Not really sure if that does anything. Be an interesting thing to test but I think the other constructor is cleaner. Other classes in java.lang use it to construct Strings.